ttl/ttlworker/scan.go

```diff
@@ -272,7 +280,9 @@ func (w *ttlScanWorker) loop() error {
 	return nil
 }
 
-func (w *ttlScanWorker) handleScanTask(ctx context.Context, task *ttlScanTask) {
+func (w *ttlScanWorker) handleScanTask(task *ttlScanTask) {
```

Comment:
It seems to be a tough problem for go context: https://github.com/golang/go/issues/36503 :cry:

Reply:
> A simple way is that we can just put the ctx check at the beginning of the for loop. In this way, tough the doScan will not exit immediately, it will exit once one sql finished.

Yep. I temporarily fix this through this way. Then the `cancel` may not work as expected. I just left a comment for it.